### ISSUE TITLE
invite_state from /sync does not contain membership for the invitee

### ISSUE THREAD

**Issue by Half-Shot** [matrix-org/dendrite#1014]
Or basically, the invite is missing from the `invite_state`, which crashes Riot.

Example invite from @Half-Shot:half-shot.uk -> @half-shot:den.half-shot.uk

```json
{
  "invite_state": {
    "events": [
      {
        "content": {
          "avatar_url": "mxc://half-shot.uk/6fc3fb176d5246f1e7cf7d89ad4248b8",
          "displayname": "halfy",
          "membership": "join"
        },
        "sender": "@Half-Shot:half-shot.uk",
        "state_key": "@Half-Shot:half-shot.uk",
        "type": "m.room.member"
      },
      {
        "content": {
          "join_rule": "invite"
        },
        "sender": "@Half-Shot:half-shot.uk",
        "state_key": "",
        "type": "m.room.join_rules"
      },
      {
        "content": {
          "alias": "#secret:half-shot.uk"
        },
        "sender": "@Half-Shot:half-shot.uk",
        "state_key": "",
        "type": "m.room.canonical_alias"
      },
      {
        "content": {
          "name": "🐶🦊🦄🐱🏡"
        },
        "sender": "@Half-Shot:half-shot.uk",
        "state_key": "",
        "type": "m.room.name"
      }
    ]
  }
}
```

**Comment by Kegsay** [comment]
Is this still valid @neilalexander ?

**Comment by neilalexander** [comment]
It was half-fixed before and will be fully fixed by #1273.
